p2p/transport/quicreuse/connmgr.go

```diff
@@ -47,17 +49,23 @@ func NewConnManager(statelessResetKey quic.StatelessResetKey, opts ...Option) (*
 	}
 
 	quicConf := quicConfig.Clone()
-	quicConf.StatelessResetKey = &statelessResetKey
 
 	var tracers []quiclogging.Tracer
-	if qlogTracer != nil {
-		tracers = append(tracers, qlogTracer)
-	}
+
 	if cm.enableMetrics {
-		tracers = append(tracers, newMetricsTracer())
+		cm.metricsTracer = newMetricsTracer()
 	}
 	if len(tracers) > 0 {
```

Comment:
Is `len(tracers)` always 0?

Reply:
maybe you want: 
```
tracers = append(tracers, cm.metricsTracer)
```

Also if enableMetrics is false, we want to enable qLogTracer initialised from 
`qloggerForDir` given that we had these lines before

```
if qlogTracer != nil {
		tracers = append(tracers, qlogTracer)
	}
```
and qlogTracer was initialised in `init`

Reply:
`github.com/quic-go/quic-go/qlog` no longer includes a `NewTracer`. This is only a connection tracer so we don't need it here anymore. It's now done below in the connection tracers.

Reply:
ah sorry I see what I mean, I think this is a half finished refactor actually.